api/scpca_portal/models/project.py

```diff
@@ -1039,6 +1044,14 @@ def update_spatial_data(future):
             f"Processing {samples_count} sample{pluralize(samples_count)} using "
             f"{max_workers} worker{pluralize(max_workers)}"
         )
+
+        # Prepare a threading.Lock for each multiplexed sample that shares a zip file.
+        # The keys are the sample.multiplexed_ids since that will be unique across shared zip files.
+        multiplexed_ids = set(
+            ["_".join(s.multiplexed_ids) for s in samples if s.has_multiplexed_data]
```

Comment:
Before performing the join, we should probably sort all of `s.multiplexed_ids` to guarantee the uniqueness of the set. It could be that the ids are sorted elsewhere, but we could remove the dependency by sorting here again (it's only a few ids in any case).

Reply:
These are already sorted so we should be good. The set is just do we don't create multiple keys.